src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIResponsesChatClient.cs

```diff
@@ -546,6 +585,45 @@ internal static FunctionTool ToResponseTool(AIFunctionDeclaration aiFunction, Ch
             aiFunction.Description);
     }
 
+    internal static ImageGenerationTool ToImageResponseTool(HostedImageGenerationTool imageGenerationTool)
+    {
+        ImageGenerationOptions? imageGenerationOptions = imageGenerationTool.Options;
+
+        // Not every option is available on ImageGenerationOptions, so we allow the tool to define a factory for 
+        // the OpenAi ImageGenerationTool.  This is important to set partial_images, input_image_mask, etc.
+        var result = imageGenerationTool.RawRepresentationFactory?.Invoke(null) as ImageGenerationTool ?? new();
+
+        // Model: Image generation model
+        if (imageGenerationOptions?.ModelId is not null && result.Model is null)
+        {
+            result.Model = imageGenerationOptions.ModelId;
+        }
+
+        // Size: Image dimensions (e.g., 1024x1024, 1024x1536)
+        if (imageGenerationOptions?.ImageSize is not null && result.Size is null)
+        {
+            // Use a custom type to ensure the size is formatted correctly.
```

Comment:
I don't understand this comment. What is the "custom type" being used? Also, the constructor for ImageGenerationToolSize states:
```
    /// <b>Note:</b> arbitrary dimensions are not supported and a given model will only support a set of predefined
    /// sizes. If supported dimensions are not known, try using one of the static properties like <see cref="W1024xH1024"/>.
```
Is it going to be likely that a size the user provides won't work? Do we know what sizes will work and should be normalize to them? Or it's model-specific and we have no way of knowing what's valid?